hub/integrations/pytorch.py

```diff
@@ -8,36 +10,52 @@
 import numpy as np
 from itertools import repeat
 from collections import defaultdict
-from typing import Any, Callable, List, Optional, Set, Dict
-from hub.util.exceptions import ModuleNotInstalledException
+from typing import Any, Callable, List, Optional, Set, Dict, Union
+from hub.util.exceptions import DatasetUnsupportedPytorch, ModuleNotInstalledException
 from hub.util.shared_memory import (
     remove_shared_memory_from_resource_tracker,
     clear_shared_memory,
 )
 from pathos.pools import ProcessPool  # type: ignore
-from hub.core.storage import MemoryProvider
 
 try:
     from multiprocessing.shared_memory import SharedMemory  # type: ignore
 except ModuleNotFoundError:
     pass
 
-_hub_storage_provider = MemoryProvider()
+from functools import lru_cache
+
+
+@lru_cache()
+def get_s3_storage(state: tuple) -> S3Provider:
+    """Ensures that s3 clients aren't initialized over and over again in the same process"""
+    s3 = S3Provider.__new__(S3Provider)
+    s3.__setstate__(state)
+    return s3
+
 
 # TODO make this use shared memory to make on the fly transforms faster. Currently using transform slows us down by 10x
```

Comment:
Please add this todo to jira bugs/backlog, if not added.

Reply:
It's already on jira.